libs/dh/esett/feature-outgoing-messages/src/lib/details.component.ts

```diff
@@ -72,96 +81,96 @@ import { DhResolveModalComponent } from './resolve.componet';
+  private httpClient = inject(HttpClient);
+  private toastService = inject(WattToastService);
+  private modalService = inject(WattModalService);
+  private navigation = inject(DhNavigationService);
+  private outgoingMessageByIdDocumentQuery = lazyQuery(GetOutgoingMessageByIdDocument);
 
-  private subscription?: Subscription;
+  outgoingMessage = computed(
+    () => this.outgoingMessageByIdDocumentQuery.data()?.esettOutgoingMessageById
+  );
 
-  outgoingMessage: DhOutgoingMessageDetailed | undefined = undefined;
+  canResolve = computed(
+    () =>
+      this.outgoingMessage()?.documentStatus === DocumentStatus.Rejected ||
+      (this.outgoingMessage()?.documentStatus === DocumentStatus.AwaitingReply &&
+        dayjs(new Date()).diff(this.outgoingMessage()?.lastDispatched, 'hours') >= 12)
+  );
 
-  @ViewChild(WattDrawerComponent)
-  drawer: WattDrawerComponent | undefined;
+  drawer = viewChild.required(WattDrawerComponent);
 
-  closed = output<void>();
+  // Router param
+  id = input.required<string>();
 
   dispatchDocument = signal<string | undefined>(undefined);
   responseDocument = signal<string | undefined>(undefined);
 
-  get messageTypeValue(): string {
-    if (this.outgoingMessage?.calculationType) {
-      return translate(
-        'eSett.outgoingMessages.shared.calculationType.' + this.outgoingMessage?.calculationType
-      );
+  loading = this.outgoingMessageByIdDocumentQuery.loading;
+
+  get messageTypeValue() {
+    const calculationType = this.outgoingMessage()?.calculationType;
+    if (calculationType) {
+      return translate('eSett.outgoingMessages.shared.calculationType.' + calculationType);
     }
 
     return emDash;
   }
 
-  public open(outgoingMessageId: string): void {
-    this.drawer?.open();
+  constructor() {
+    afterRenderEffect(() => {
+      this.drawer().open();
+      const id = this.id();
 
-    this.loadOutgoingMessage(outgoingMessageId);
-  }
+      if (id) {
+        this.outgoingMessageByIdDocumentQuery.query({ variables: { documentId: id } });
+      }
+    });
 
-  onClose(): void {
-    this.closed.emit();
+    afterRenderEffect(() => {
+      const outgoingMessage = this.outgoingMessage();
+
+      if (outgoingMessage === undefined) {
+        return;
+      }
+
+      if (
+        outgoingMessage.documentId &&
+        outgoingMessage.documentStatus !== DocumentStatus.Received
+      ) {
+        this.loadDocument(outgoingMessage.dispatchDocumentUrl, this.dispatchDocument.set);
+      }
+
+      if (
+        outgoingMessage.documentId &&
+        ((outgoingMessage.documentStatus !== DocumentStatus.Received &&
+          outgoingMessage.documentStatus === DocumentStatus.Accepted) ||
+          outgoingMessage.documentStatus === DocumentStatus.Rejected ||
+          outgoingMessage.documentStatus === DocumentStatus.ManuallyHandled)
+      ) {
+        this.loadDocument(outgoingMessage.responseDocumentUrl, this.responseDocument.set);
+      }
```

Comment:
suggestion: perfect candidate for `resource`/`rxResource`? 

https://angular.dev/guide/signals/resource
https://angular.dev/api/core/rxjs-interop/rxResource 

Can be a follow-up PR.

Reply:
Good idea, would be nice to use them